source/common/http/filter/fault_filter.cc

```diff
@@ -142,6 +150,21 @@ void FaultFilter::abortWithHTTPStatus() {
   callbacks_->requestInfo().setResponseFlag(Http::AccessLog::ResponseFlag::FaultInjected);
 }
 
+bool FaultFilter::matchesTargetCluster(HeaderMap& headers) {
+  bool matches = true;
+
+  if (!config_->targetCluster().empty()) {
+    // TODO PERF Eliminate repeated route matching
```

Comment:
And ratelimit filter as well. For example, if you have a fault injection filter and a rate limit filter (each looking up the upstream cluster), then you are looking at 3 route lookups for each request.